src/main/java/org/opensearch/forecast/model/Forecaster.java

```diff
@@ -0,0 +1,410 @@
+        Integer shingleSize = null;
+        List<Feature> features = new ArrayList<>();
+        Integer schemaVersion = CommonValue.NO_SCHEMA_VERSION;
+        Map<String, Object> uiMetadata = null;
+        Instant lastUpdateTime = null;
+        User user = null;
+        String resultIndex = null;
+
+        List<String> categoryField = null;
+        Integer horizon = null;
+        ImputationOption interpolationOption = null;
+
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String fieldName = parser.currentName();
+            parser.nextToken();
+
+            switch (fieldName) {
+                case NAME_FIELD:
+                    name = parser.text();
+                    break;
+                case DESCRIPTION_FIELD:
+                    description = parser.text();
+                    break;
+                case TIMEFIELD_FIELD:
+                    timeField = parser.text();
+                    break;
+                case INDICES_FIELD:
+                    ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.currentToken(), parser);
+                    while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                        indices.add(parser.text());
+                    }
+                    break;
+                case UI_METADATA_FIELD:
+                    uiMetadata = parser.map();
+                    break;
+                case CommonName.SCHEMA_VERSION_FIELD:
+                    schemaVersion = parser.intValue();
+                    break;
+                case FILTER_QUERY_FIELD:
+                    ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+                    try {
+                        filterQuery = parseInnerQueryBuilder(parser);
+                    } catch (ParsingException | XContentParseException e) {
+                        throw new ValidationException(
+                            "Custom query error in data filter: " + e.getMessage(),
+                            ValidationIssueType.FILTER_QUERY,
+                            ValidationAspect.DETECTOR
+                        );
+                    } catch (IllegalArgumentException e) {
+                        if (!e.getMessage().contains("empty clause")) {
+                            throw e;
+                        }
+                    }
+                    break;
+                case FORECAST_INTERVAL_FIELD:
+                    try {
+                        forecastInterval = TimeConfiguration.parse(parser);
+                    } catch (Exception e) {
+                        if (e instanceof IllegalArgumentException && e.getMessage().contains(CommonMessages.NEGATIVE_TIME_CONFIGURATION)) {
+                            throw new ValidationException(
+                                "Detection interval must be a positive integer",
```

Comment:
message should be changed to "Forecasting Interval" and ValidationAspect to FORECASTER.

Reply:
Good catch. Changed.